_search-plugins/knn/performance-tuning.md

```diff
@@ -98,6 +98,57 @@ In OpenSearch 2.15 or later, you can further improve indexing speed and reduce d
 This is an expert-level setting. Disabling the `_recovery_source` may lead to failures during peer-to-peer recovery. Before disabling the `_recovery_source`, check with your OpenSearch cluster admin to determine whether your cluster performs regular flushes before starting the peer-to-peer recovery of shards before disabling the `_recovery_source`.  
 {: .warning}
 
+### (Expert-level) Build vector data structures on demand
+
+This approach is recommended only for workloads that involve a single initial bulk upload and will be used exclusively for search after force merging to a single segment.
+
+During indexing, vector search builds a specialized data structure for a `knn_vector` field to enable efficient approximate k-NN search. However, these structures are rebuilt from scratch during [force merge]({{site.url}}{{site.baseurl}}/api-reference/index-apis/force-merge/) for k-NN indexes. To optimize indexing speed, follow these steps:
+
+1. **Disable vector data structure creation**: Disable building vector data structures for new segments by setting [`index.knn.advanced.approximate_threshold`]({{site.url}}{{site.baseurl}}/search-plugins/knn/knn-index/#index-settings) to `-1` either at index creation or at any time after that:
+
+    ```json
+    PUT /test-index/_settings
+    {
+      "index.knn.advanced.approximate_threshold": "-1"
+    }
+    ```
+    {% include copy-curl.html %}
+
+1. **Perform bulk indexing**: Index data in [bulk]({{site.url}}{{site.baseurl}}/api-reference/document-apis/bulk/) without performing any searches during ingestion:
+
+    ```json
+    POST _bulk
+    { "index": { "_index": "test-index", "_id": "1" } }
+    { "my_vector1": [1.5, 2.5], "price": 12.2 }
+    { "index": { "_index": "test-index", "_id": "2" } }
+    { "my_vector1": [2.5, 3.5], "price": 7.1 }
+    ```
+    {% include copy-curl.html %}
+
+    If searches are performed while vector data structures are disabled, they will run using exact k-NN search.
+
+1. **Reenable vector data structure creation**: Once indexing is complete, enable vector data structure creation by setting `index.knn.advanced.approximate_threshold` to `0`:
+
+    ```json
+    PUT /test-index/_settings
+    {
+      "index.knn.advanced.approximate_threshold": "0"
+    }
+    ```
+    {% include copy-curl.html %}
+
+    If you forget to reset the setting to `0` before the force merge, you will need to reindex your data.
+    {: .note}
+
+1. **Force merge segments into one segment**: Perform a force merge and specify `max_num_segments=1` to create the vector data structures only once:
+
+    ```json
+    POST test-index/_forcemerge?max_num_segments=1
+    ```
+    {% include copy-curl.html %}
+
+    After the force merge, with vector data structures built, new search requests will execute approximate k-NN search.
```

Comment:
"After the force merge, new search requests will execute approximate k-NN search using the newly created data structures"?